express-notes-api/index.js

```diff
@@ -0,0 +1,76 @@
+
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const port = 3000;
+const data = require('./data.json');
+
+app.use(express.json());
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  for (const id in data.notes) {
+    arr.push(data.notes[id]);
+  }
+  res.status(200).send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (req.params.id < 0) {
+    res.status(400).send('Error: Number must be positive');
+  }
+  if (data.notes[req.params.id]) {
+    res.status(200).send(data.notes[req.params.id]);
+  } else {
+    res.status(404).send(`Error: ${req.params.id} cannot be found`);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  if (Object.keys(req.body)[0] === 'content') {
+    data.notes[data.nextId] = req.body;
+    data.notes[data.nextId].id = data.nextId;
+    data.nextId++;
+    fs.writeFile('./data.json', JSON.stringify(data, null, 2), 'utf8', err => {
+      if (err) throw err;
```

Comment:
Instead of throwing (which will crash your server), log the error for debugging purposes and then send a `500` response. Apply this in your other uses of writeFile too.

```js
if (err) {
  console.error(err);
  res.status(500).json({ error: 'an unexpected error occurred' })
} else {
  // ...
}
```